engine/engine.c

```diff
@@ -227,7 +227,7 @@ void engine_init (const char *const pwd_filename, int do_not_open_port) {
 
   if (E->settings_addr.s_addr) {
     ipv4 = ntohl (E->settings_addr.s_addr);
-    if ((ipv4 >> 24) != 10) {
```

Comment:
Honestly I don't think this check should be there at all, but, according to original devs, that code is correct, "even if it's illogical in terms of mtproxy".